src/app/Notifications/Notifications.tsx

```diff
@@ -57,12 +58,35 @@ export class Notifications {
   private readonly _notifications$: BehaviorSubject<Notification[]> = new BehaviorSubject<Notification[]>(
     this._notifications
   );
+  private readonly _drawerState$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  constructor() {
+    this._drawerState$.pipe(filter((v) => v)).subscribe(() =>
+      this._notifications$.next(
+        this._notifications.map((n) => {
+          n.hidden = true;
+          return n;
+        })
+      )
+    );
+  }
+
+  drawerState(): Observable<boolean> {
+    return this._drawerState$.asObservable();
+  }
+
+  setDrawerState(state: boolean): void {
+    this._drawerState$.next(state);
+  }
 
   notify(notification: Notification): void {
     if (!notification.key) {
       notification.key = nanoid();
     }
     notification.read = false;
+    if (notification.hidden === undefined) {
```

Comment:
Some reason behind this checking for `undefined`?

Reply:
Not currently, but I thought it could be useful in the future. This would allow for a notification to be emitted with `hidden: true` explicitly set by the caller so that the notification doesn't generate a toast, even if the drawer is closed. This way the drawer state is only used as the default visibility setting if the caller hasn't specified one.

Reply:
Ahh right cool! Makes sense :D

Reply:
I guess the websocket connection notification would be a good example of such a "quiet" notification though. May as well do it now.